astropy/io/votable/converters.py

```diff
@@ -842,6 +828,12 @@ class Double(FloatingPoint):
 
     format = "f8"
 
+    def binparse(self, read):
```

Comment:
Seems a little repetitive across the board. Looks like the only difference is a number. Can this method be inherited but access some `self._expected_len` (name negotiable) that is set by the subclass?

Reply:
Great suggestion, looking back at it now I can see there is a lot of duplication. I've moved things up to the parent Numeric class, let me know if that looks good. (Or if you see any adjustments needing to be made)